### PR TITLE
Avoid allocating on the heap

### PR DESCRIPTION
**Pull request by markand**

### CODE REVIEW

File: ttyclock.c [hunk 1]
```diff
@@ -61,18 +61,19 @@ init(void)
...
+     init_pair(0, ttyclock.bg, ttyclock.bg);
+     init_pair(1, ttyclock.bg, ttyclock.option.color);
+     init_pair(2, ttyclock.option.color, ttyclock.bg);
+//     init_pair(0, ttyclock.bg, ttyclock.bg);
+//     init_pair(1, ttyclock.bg, ttyclock.option.color);
+//     init_pair(2, ttyclock.option.color, ttyclock.bg);
      refresh();
 
      /* Init signal handler */
+     memset(&sig, 0, sizeof (sigaction));
```
Comment (anarcat) [inline ttyclock.c]: is this really related to the `ttyclock` object change? or should this be split out? it sure doesn't seem related to the PR's description, at least not directly: `sig` is already on the stack, FWICT.

**Comment by anarcat** [comment]
oh and also, if you are going to reroll a commit, please make it explicit "why" (as opposed to "what") this commit is necessary. it may seem obvious to you, but newcomers or passers by would benefit for a more explicit justification.

thanks again

File: ttyclock.c (hunk 1 above)
Comment (markand) [reply ttyclock.c]: Done.

File: ttyclock.c [hunk 2]
```diff
@@ -166,15 +166,10 @@ signal_handler(int signal)
...
-    if (ttyclock && ttyclock->tty)
-        free(ttyclock->tty);
-    if (ttyclock && ttyclock->option.format)
-        free(ttyclock->option.format);
-    if (ttyclock)
-        free(ttyclock);
+    if (ttyclock.ttyscr)
+        delscreen(ttyclock.ttyscr);
+
+    free(ttyclock.tty);
```
Comment (anarcat) [inline ttyclock.c]: we dropped the tty check here... shouldn't this be:

```suggestion
    if (ttyclock.tty)
        free(ttyclock.tty);
```

File: ttyclock.h [hunk 3]
```diff
@@ -78,7 +78,7 @@ typedef struct
           Bool utc;
           Bool box;
           Bool noquit;
-          char *format;
+          char format[512];
```
Comment (anarcat) [inline ttyclock.h]: it seems we gratuitously changed the length of the format string from 100 to 512 here.

**Comment by anarcat** [comment]
> Is it okay now?

not quite, but very close! :) thanks again for your patience.

File: ttyclock.c (hunk 2 above)
Comment (markand) [reply ttyclock.c]: The `free` function does not need any check.

File: ttyclock.h (hunk 3 above)
Comment (markand) [reply ttyclock.h]: Should I revert to 100?

Comment (markand) [reply ttyclock.h]: Yes, I think it's large enough for a date.
